packages/clay-time-picker/src/index.tsx

```diff
@@ -4,57 +4,422 @@
+	config?: {
+		use12Hours: Config;
+		use24Hours: Config;
+	};
+	icon?: boolean;
+	onInputChange: (values: Input) => void;
+	values: Input;
+	spritemap?: string;
 	timezone?: string;
+	use12Hours?: boolean;
+	useNative?: boolean;
 }
 
-const timePickerContent: FunctionComponent<ITPContentProps> = ({
-	defaultValue,
-	onChange,
-	onInput,
+const KEY_BACKSPACE = 8;
+const KEY_ARROWRIGHT = 39;
+const KEY_ARROWLEFT = 37;
+const KEY_ARROWUP = 38;
+const KEY_ARROWDOWN = 40;
+const DEFAULT_VALUE = '--';
+const DEFAULT_CONFIG = {
+	use12Hours: {
+		ampm: {
+			am: 'AM',
+			pm: 'PM',
+		},
+		hours: {
+			max: 11,
+			min: 0,
+		},
+		minutes: {
+			max: 59,
+			min: 0,
+		},
+	},
+	use24Hours: {
+		ampm: {
+			am: 'AM',
+			pm: 'PM',
+		},
+		hours: {
+			max: 23,
+			min: 0,
+		},
+		minutes: {
+			max: 59,
+			min: 0,
+		},
+	},
+};
+
+const ClayTimePicker: React.FunctionComponent<IProps> = ({
+	config = DEFAULT_CONFIG,
 	timezone,
-	value,
-	...otherProps
-}) => (
-	<React.Fragment>
-		<div className="input-group-item">
-			<input
-				{...otherProps}
-				className="form-control"
-				data-testid="time-picker-input"
-				defaultValue={defaultValue}
-				onChange={onChange}
-				onInput={onInput}
-				type="time"
-				value={value}
-			/>
-		</div>
-		{timezone && (
-			<div className="input-group-item input-group-item-shrink">
-				<span className="input-group-text">{`(${timezone})`}</span>
-			</div>
-		)}
-	</React.Fragment>
-);
+	icon = false,
+	spritemap,
+	use12Hours = false,
+	values = {
+		ampm: DEFAULT_VALUE,
+		hours: DEFAULT_VALUE,
+		minutes: DEFAULT_VALUE,
+	},
+	onInputChange = () => {},
+}) => {
+	const useConfig: Config = config[use12Hours ? 'use12Hours' : 'use24Hours'];
+	const [actionVisible, setActionVisible] = useState(false);
+	const [isFocused, setIsFocused] = useState();
+	const elementRef = useRef<null | HTMLDivElement>(null);
+	const hoursRef = useRef<null | HTMLInputElement>(null);
+	const minutesRef = useRef<null | HTMLInputElement>(null);
+	const ampmRef = useRef<null | HTMLInputElement>(null);
+	const defaultElementFocus = {
+		configName: TimeType.hours,
+		focused: false,
+	};
+	const elementFocusedRef = useRef<{
+		configName: TimeType;
+		focused: boolean;
+	}>(defaultElementFocus);
+	const regex = /^\d+$/;
 
-interface IProps extends ITPContentProps {
-	wrapTime?: boolean;
-}
+	const handleMaxAndMin = (value: string, config: ConfigMaxMin) => {
+		const newValue = value.substring(value.length - 2, value.length);
+		const intrinsicValue = Number(newValue);
 
-const ClayTimePicker: FunctionComponent<IProps> = ({
-	wrapTime = true,
-	...otherProps
-}) => {
-	if (wrapTime) {
-		return (
-			<div className="input-group">{timePickerContent(otherProps)}</div>
-		);
-	}
+		if (intrinsicValue > config.max) {
+			return String(config.min);
+		} else if (intrinsicValue < config.min) {
+			return String(config.max);
+		}
+
+		return newValue;
+	};
+
+	const handleKeyDown = (
```

Comment:
It might beneficial to abstract this into a hook. Probably not needed for this PR, but it might be nice in the future.

Reply:
I'm not so sure, maybe we can create a hook that can handle focus, something like I'm seeing being done at https://github.com/facebook/react/pull/15849, I'm trying to get close to that with #1987 , but we will probably still maintain this function because we have things unique only to that component.

Reply:
ah gotcha, sounds good.